ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFFromUnixTime.java

```diff
@@ -87,29 +75,19 @@ public ObjectInspector initialize(ObjectInspector[] arguments) throws UDFArgumen
         inputLongOI = (LongObjectInspector) arguments[0];
         break;
       default:
-        throw new UDFArgumentException("The function " + getName().toUpperCase()
+        throw new UDFArgumentException("The function " + getFuncName().toUpperCase()
             + " takes only int/long types for first argument. Got Type:" + arg0OI.getPrimitiveCategory().name());
     }
 
     if (arguments.length == 2) {
-      PrimitiveObjectInspector arg1OI = (PrimitiveObjectInspector) arguments[1];
-      switch (arg1OI.getPrimitiveCategory()) {
-        case CHAR:
-        case VARCHAR:
-        case STRING:
-          inputTextConverter = ObjectInspectorConverters.getConverter(arg1OI,
-              PrimitiveObjectInspectorFactory.javaStringObjectInspector);
-          break;
-        default:
-          throw new UDFArgumentException("The function " + getName().toUpperCase()
-              + " takes only string type for second argument. Got Type:" + arg1OI.getPrimitiveCategory().name());
-      }
+      checkArgGroups(arguments, 1, inputTypes, STRING_GROUP);
+      obtainStringConverter(arguments, 1, inputTypes, converters);
     }
 
     if (timeZone == null) {
```

Comment:
remove  timeZone == null check. Always override the timeZone variable with value stored in session or default value.